graph/builder.go

```diff
@@ -666,30 +660,32 @@ func (b *Builder) pruneZombieChans() error {
 
 // handleNetworkUpdate is responsible for processing the update message and
 // notifies topology changes, if any.
-//
-// NOTE: must be run inside goroutine.
-func (b *Builder) handleNetworkUpdate(update *routingMsg) {
-	defer b.wg.Done()
+func (b *Builder) handleNetworkUpdate(msg any,
+	op ...batch.SchedulerOption) error {
+
+	select {
+	case <-b.quit:
+		return ErrGraphBuilderShuttingDown
+	default:
+	}
 
 	// Process the routing update to determine if this is either a new
 	// update from our PoV or an update to a prior vertex/edge we
 	// previously accepted.
 	var err error
-	switch msg := update.msg.(type) {
+	switch msg := msg.(type) {
 	case *models.LightningNode:
-		err = b.addNode(msg, update.op...)
+		err = b.addNode(msg, op...)
 
 	case *models.ChannelEdgeInfo:
-		err = b.addEdge(msg, update.op...)
+		err = b.addEdge(msg, op...)
```

Comment:
It seems to me that the original reason for adding concurrency might have been due to `addEdge` being relatively slow. I wonder if that was indeed the case 🤔. It could be worth running a quick benchmark by syncing the entire graph. wdyt?

Reply:
so this isnt changing anything as it stands today. It may look like it is being done asynchronously today because of the go routine but it is not cause the only path is from the public `AddEdge` which waits for the response:

```
func (b *Builder) AddEdge(edge *models.ChannelEdgeInfo,
	op ...batch.SchedulerOption) error {

	rMsg := &routingMsg{
		msg: edge,
		op:  op,
		err: make(chan error, 1),
	}

	select {
	case b.networkUpdates <- rMsg:
		select {
		case err := <-rMsg.err:
			return err
		case <-b.quit:
			return ErrGraphBuilderShuttingDown
		}
	case <-b.quit:
		return ErrGraphBuilderShuttingDown
	}
}
```

Reply:
the only part that is _actually_ done asynchronously is the graph topology stuff as that happens after the write to the `rMsg.err` channel. And that part is kept asynchronous here

Reply:
>  the original reason for adding concurrency 

the original origin of why this was like this afaik is cause initially all this code was part of the gossiper's code (which has the exact pattern (same naming even)) and in that area, things are actually done concurrently. 

But then when things were moved here, we copied the pattern but then needed to make it concurrent so that the gossiper could call this code and know if it succeeded/failed

Reply:
Ah right, you're correct! Thanks for the added explanation 🙏